src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,69 @@
+import TransparentButton from "components/Buttons/TransparentButton";
+import "./Pagination.css";
+
+type Props = {
+  page: number;
+  pageCount: number;
+  onPageChange: any;
+};
+
+function range(start, end) {
+  let length = end - start + 1;
+  return Array.from({ length }, (_, idx) => idx + start);
+}
+
+function getPageNumbers(current, max = 1) {
+  if (max === 2) {
+    return [1, 2];
+  }
+  if (current <= 1) {
+    return range(current, current + 2);
+  }
+
+  if (current + 1 <= max) {
+    return range(current - 1, current + 1);
+  }
+
+  if (current <= max) {
+    return range(current - 2, current);
+  }
+  return [];
+}
```

Comment:
if i understand correctly the goal is to display previous, current and next pages in UI. i think function can be simplified to:

```
const ret = [];
if (current > 1) {
  ret.push(current - 1};
}
ret.push(current);
if (current < max) {
  ret.push(current + 1);
}
return ret;
```

Reply:
No the requirements was a bit different. So we wanted to have three elements if possible to maintain the gas as you can see in the image below.

<img width="421" alt="Screenshot 2023-01-12 at 11 48 02 AM" src="https://user-images.githubusercontent.com/73279781/211992513-62b96129-4644-44db-9124-a2fc1c106df9.png">

The function need to fulfil the following requirements:

if max is 1 return empty
if current is 1 and max is 2 return 1,2
if current is 2 and max is 2 return 1,2
if current is 1 and max is 4 return 1,2,3
if current is 2 and max is 4 return 1,2,3
if current is 3 and max is 4 return 2,3,4
if current is 4 and max is 4 return 2,3,4

The above solution will return 2 numbers in certain situations like current is 1 and max is 4 it will return 1 and 2. But it should return 1, 2, 3.

BTW I have refactored the code to this:

```js
function getPageNumbers(current, max = 1) {
  if (max === 1) return [];
  if (current === 1) {
    return max >= 3 ? [1, 2, 3] : [1, 2];
  } else if (current === max) {
    return max >= 3 ? [current - 2, current - 1, current] : [current - 1, current];
  } else {
    return [current - 1, current, current + 1];
  }
}
```

Reply:
okay, got it